sat/cli/bootsys/bos.py

```diff
@@ -277,7 +277,7 @@ def has_completed(self):
             return False
 
         except APIError as err:
-            LOGGER.warning('Failed to query session status: %s', err)
+            raise WaitingFailure('Failed to query session status: %s', err)
```

Comment:
The string must be formatted before passing into the exception class. E.g.:

```suggestion
            raise WaitingFailure(f'Failed to query session status: {err}')
```